stock_restrict_lot/models/stock_picking.py

```diff
@@ -0,0 +1,10 @@
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    restrict_lot_id = fields.Many2one(
```

Comment:
Shouldn't be a o2m so called `restrict_lot_ids` ?

Reply:
The only purpose is to be able to search, and it works fine this way.
Here is the same kind of stuff : https://github.com/odoo/odoo/blob/16.0/addons/stock/models/stock_picking.py#L410

Reply:
Even odoo is doing that, i would prefer a proper definition if it works as well, if some one else (a developer) is reusing this related field it makes hard to expect it returns a recordset instead a record.

Reply:
I usually avoid doing stuff differently than Odoo SA when it works well to avoid unexpected issues.
I could add a comment to indicates that this is a technical field which purpose is to be able to search a picking by restrict lot.
Actually I don't even know how it works technically for the search to work with this m2o not stored field.

Anyway, I just made a test with a compute o2m (the goal is to keep a not stored field since we don't need this field for any other purpose that the search, at least for now) and the filter does not work.

`2023-03-08 09:31:30,426 165 ERROR db odoo.osv.expression: Non-stored field stock.picking.restrict_lot_ids cannot be searched.`

I think it is reasonable to go with the current solution for a start, do you agree?

Reply:
ok then, good to know thanks for testing that, leaving a comment seems a good idea !